Extensions/ArtifactEngine/Engine/cilogger.ts

```diff
@@ -1,21 +1,45 @@
+import { error } from "util";
+
 var tl = require('vsts-task-lib');
+var packagejson = require('../package.json');
 
 const area: string = 'artifact-engine';
 
 function getDefaultProps() {
     return {
-        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        serverurl: tl.getVariable('System.TEAMFOUNDATIONSERVERURI'),
+        releaseurl: tl.getVariable('Release.ReleaseWebUrl'),
         releaseid: tl.getVariable('Release.ReleaseId'),
         builduri: tl.getVariable('Build.BuildUri'),
-        buildid: tl.getVariable('Build.Buildid')
+        buildid: tl.getVariable('Build.Buildid'),
+        agentVersion: tl.getVariable('Agent.Version'),
+        version: packagejson.version
     };
 }
 
-export function publishEvent(feature, properties: { [key: string]: any }): void {
+export function publishEvent(feature, properties: any | IReliabilityData): void {
```

Comment:
why not only IReliabilityData?

Reply:
the same method is used to publish perf data also, I didn't create an interface for it since it doesn't seem much use currently.